apps/core/src/config/environment.ts

```diff
@@ -0,0 +1,18 @@
+const ENV_VAR_UNDEFINED = 'undefined';
+
+/**
+ * Parses the environment variable into a string if the variable is defined; otherwise, throws the given error.
+ * @param variable the environment variable to parse
+ * @param error the given error to throw if the variable is not defined
+ * @returns the environment variable in string type
+ */
+export const sanitizeEnvVarOrThrow = (
+  variable: string | undefined,
+  error: Error,
+): string => {
+  if (variable === undefined || variable === ENV_VAR_UNDEFINED) {
+    throw error;
+  }
+
+  return variable;
+};
```

Comment:
Nice little utility!

Reply:
Non-blocking suggestion: One thing about functions that throw is TS doesn't make you catch the error. Instead returning `undefined | string` will make TS force you to handle the failure case.

Reply:
Non-blocking suggestion: I highly recommend we start using https://github.com/alexreardon/tiny-invariant anywhere we claim to know more than the compiler (i.e., we know a value is not null, but it technically is optional. Using tiny-invariant, you make a promise to the compiler that if you're wrong, it can throw. This is so much better than assertions.

Not that this necessarily applies to this particular function. I just want to suggest it's usage where you can.

Reply:
the [tiny-invariant](https://github.com/alexreardon/tiny-invariant) is pretty cool
it would apply to many of the falsey checks that we do 👍

Reply:
> Non-blocking suggestion: One thing about functions that throw is TS doesn't make you catch the error. Instead returning `undefined | string` will make TS force you to handle the failure case.

Trying to use this as syntactic sugar to throw for missing environment variables and stop the server from starting. If we return `undefined | string`, we will have to throw at the caller function and render this util useless.